src/main.py

```diff
@@ -178,6 +185,60 @@ def run_plan_bot(
             skip_tracking=skip_tracking,
         )
 
+    params = {}
+    if comments_progress:
+        newest_comment_id = comments_progress.get("newest")
+        oldest_comment_id = comments_progress.get("oldest")
+
+        if newest_comment_id:
+            params["before"] = latest_comment_id
+
+    # Get comments newer than latest
+    current_newest_comment_id = None
+    for comment in subreddit.comments(params=params):
+        comment = reddit_util.Comment(comment)
+        process_post(
+            comment,
+            plans,
+            posts_db,
+            post_ids_processed,
+            send=send_replies,
+            simulate=simulate_replies,
+            skip_tracking=skip_tracking,
+        )
+        if not current_newest_comment_id:
+            current_newest_comment_id = comment.fullname
+            if not skip_tracking:
+                comments_progress.reference.set(
```

Comment:
what does calling `.reference.set` do?

Reply:
It's [this](https://googleapis.dev/python/firestore/latest/document.html#google.cloud.firestore_v1.document.DocumentSnapshot.reference). Since I used [`.get()`](https://googleapis.dev/python/firestore/latest/document.html#google.cloud.firestore_v1.document.DocumentReference.get) on the document reference, `comments_progress` is the document snapshot. `.reference` gets back the document reference. At that point, I can call [`.set()`](https://googleapis.dev/python/firestore/latest/document.html#google.cloud.firestore_v1.document.DocumentReference.set)